src/pool-bin/libraries/BinPool.sol

```diff
@@ -317,15 +317,21 @@ library BinPool {
             bytes32 binReserves = self.reserveOfBin[id];
             uint256 supply = self.shareOfBin[id];
 
-            _subShare(self, params.from, id, params.salt, amountToBurn);
+            amountToBurn = _subShare(self, params.from, id, params.salt, amountToBurn);
 
             bytes32 amountsOutFromBin = binReserves.getAmountOutOfBin(amountToBurn, supply);
+            if (supply == amountToBurn) feeAmountToProtocol = binReserves.getAmountOutOfBin(MINIMUM_SHARE, supply);
```

Comment:
also this should be addition as we are in the loop. so feeAmountToProtocol is being overwritten if we loop twice. can you see this implementation? moved the code a bit 


```
    /// @notice Burn user's share and withdraw tokens form the pool.
    /// @return result the delta of the token balance of the pool
    function burn(State storage self, BurnParams memory params)
        internal
        returns (BalanceDelta result, uint256[] memory ids, bytes32[] memory amounts, bytes32 feeAmountToProtocol)
    {
        ids = params.ids;
        uint256 idsLength = ids.length;
        uint256[] memory amountsToBurn = params.amountsToBurn;

        if (idsLength == 0 || idsLength != amountsToBurn.length) revert BinPool__InvalidBurnInput();

        bytes32 amountsOut;
        amounts = new bytes32[](idsLength);
        for (uint256 i; i < idsLength;) {
            uint24 id = ids[i].safe24();
            uint256 amountToBurn = amountsToBurn[i];

            if (amountToBurn == 0) revert BinPool__BurnZeroAmount(id);

            bytes32 binReserves = self.reserveOfBin[id];
            uint256 supply = self.shareOfBin[id];

            // if user is last lp for this bin, amountToBurn would be amountToBurn + MINIMUM_SHARE
            amountToBurn = _subShare(self, params.from, id, params.salt, amountToBurn);

            bytes32 amountsOutFromBin = binReserves.getAmountOutOfBin(amountToBurn, supply);
            if (amountsOutFromBin == 0) revert BinPool__ZeroAmountsOut(id);

            if (supply == amountToBurn) {
                _removeBinIdToTree(self, id);

                /// @notice withdraw all the liquidity from the bin, the locked up share will be withdrawn as protocol fee
                feeAmountToProtocol = feeAmountToProtocol.add(binReserves.getAmountOutOfBin(MINIMUM_SHARE, supply));
                amountsOutFromBin.sub(feeAmountToProtocol);
            }

            binReserves = binReserves.sub(amountsOutFromBin);

            self.reserveOfBin[id] = binReserves;
            amounts[i] = amountsOutFromBin;
            amountsOut = amountsOut.add(amountsOutFromBin);

            unchecked {
                ++i;
            }
        }

        result = toBalanceDelta(amountsOut.decodeX().safeInt128(), amountsOut.decodeY().safeInt128());
    }
```